### ISSUE TITLE
Is it possible to guarantee a minimum number of search results?

### ISSUE THREAD

**Issue by haroldadmin** [lucaong/minisearch#81]
Thank you for the great work you have done on this library. I have a use-case that I'm not sure how to accomplish with `minisearch`, and would appreciate some help with it.

Is it possible to return a minimum number of search results on every query?

Something like the following API would be great:
```ts
const index = new MiniSearch<SomeEntity>({
  fields: ["name", "synonym"],
});

index.addAll(allEntities);
const results = index.search("<some-sensible-input", { minResults: 5, limit: 10 }).
// 5 <= results.length < 10
```

I want to be able to guarantee a minimum number of results from the index even if the search query is some random string. I do not care if the returned results have a very poor match score against the query.

**Comment by lucaong** [comment]
Hi @haroldadmin ,
this specific feature is not offered by `MiniSearch`, mostly because it would not "know" which results to show in case the actual search results do not meet the minimum quantity.

I would recommend implementing this in your code, so you can choose what to show. One possibility is the following:

```javascript
const index = new MiniSearch<SomeEntity>({
  fields: ["name", "synonym"],
});

index.addAll(allEntities);

// Maintain a map of entities by ID, so we can return results
const entityById = allEntities.reduce((byId, entity) => {
  byId[entity.id] = entity
  return byId
}, {})

// Keep a list of "filler" results if the search does not return enough.
// It's up to you what results to choose, here I just take the first 5 entities
const fillers = allEntities.slice(0, 5)

// Map search results to corresponding entities:
let resultEntities = index.search("<some-sensible-input").map((result) => {
  return entityById[result.id]
})

// Add fillers to the results if there is less than 5, to have at least 5 results:
if (resultEntities.length < 5) {
  resultEntities.concat(fillers.slice(0, 5 - resultEntities.length))
}

// resultEntities is now always at least 5 elements long
```

Note that I did not run this code, so you might find an occasional bug ;)

**Comment by haroldadmin** [comment]
Thank you for the fillers suggestion.

I was hoping that there's a non-zero relevance score for each entry in the search-index for every query. This could be used to return a list of results sorted in descending order according to their score, even for queries with no strong-matches. 

I am not familiar with the internals of MiniSearch, so this is just speculation.

**Comment by lucaong** [comment]
Hi @haroldadmin 
all the results with a non-zero relevance score are always returned by `search()`, but it is possible for a query to match no document at all. For example, imagine we have these documents with a single text field: `hello there`, `hi there`, and `there were many people`. If I search using exact match using the query `there`, I would find all three, with different scores (the last one for example will have a slightly lower TF-IDF score). If I would search for `something weird`, no document would match, because none of the query terms match any of the document terms.

Fuzzy and prefix match change things a bit, but even in that case it is possible for a query to match no document at all, not even by prefix or fuzzy match.

In other words, what you propose is already the case: all non-zero scored documents are returned, in order of descending score (one can always limit the results with `results.slice(0, maxLength)`). But if a query matches no document at all, there is no result with a non-zero score.

I hope this clarifies it 🙂

**Comment by haroldadmin** [comment]
Thanks, that clears it up. Appreciate the thorough responses.
